src/Resources/assets/controllers/accordion_controller.js

```diff
@@ -61,19 +54,24 @@ export default class extends Controller {
         const current = event.currentTarget;
         const header = this.closestTarget(current, 'header');
         const arrow = this.closestTarget(current, 'arrow');
-        const content = header.nextElementSibling;
+        // Subtables can be defined in the definition. It the content block outputs them, we can toggle them
```

Comment:
@marcwieland95 `definition` is just a word. Let us be very explicit on what we mean here: "can be defined in the araise definition file" or  whatever this is.

And then: "It the content block outputs them, we can toggle them". When is that the case?

Time to take 5 steps back and explain it like somebody is only looking at this code. Because I still don't understand what is happening here.

My proposal, though I don't know if that is true:

```
// If the subtable feature is activated in the araise definition file, it's possible that
// these subtables are rendered inside accordions. In this case, we want the ... yada yada

This whole change starts to make less and less sense to me because I wonder why the accordion needs to know about sub tables at all. An accordion at best has a summary element and it's content, which has a wrapper the accordion wrapps around all content. Therefore, collapsing the accordion would just toggle the wrapper visibility. And yet, for some reason with subtables, assuming they are not part of the accordion and can be set from outside, this handling seems to work differently.

So this confusion should be resolved in the code comments because that's not at all how an accordion would be made IMHO.

Reply:
@renestalder Somebody working with araise knows what a "definition" is. Because that's one of the core concepts.
In depth I also don't know the backend logic and how things are connected. Things are not that simple as activating something like a flag.

According to the subtables. Those are tables inside tables. Don't ask.
Within the inner table there can be even two tables rendered. I agree that those tables have to be placed inside a wrapper div to make the toggle work seamlessly and as expected and the accordion controller can be reused.

![CleanShot 2024-07-18 at 13 32 48@2x](https://github.com/user-attachments/assets/01b8c539-c128-4120-8d54-04787c19ab6e)

I tried to wrap it inside a div, but ended up with an empty div and the content rendered afterwards.

![CleanShot 2024-07-18 at 13 43 02@2x](https://github.com/user-attachments/assets/fa2631ae-9cd0-4698-9a9f-4caac405bec5)

At this point I just want to make it work and don't bother too much if it's perfect or not.
In future major releases we will go through those parts anyways with Twig live components and have to rewrite thinks.

Reply:
@marcwieland95 My feedback isn't targeting perfection, but code health improvements vs degradation. Adding static parts to Stimulus controllers is counterintuitive to the principles of Stimulus; therefore, it requires good documentation of the reasoning. If something later happens out of “magic” and people try to understand that “magic” they are either bothered with confusion or pleased with documentation.

The frontend code needs to be understood without prior knowledge about Symfony. This has to be maintainable for frontend developers that are not PHP developers, including myself.

So, coming back to my point: Good documentation is the minimum. We can improve the situation by taking 2 more minutes to form an understandable sentence with everything we need to know, which is helping everyone, in contrast to just assuming everyone has knowledge about the underlying workings of everything.

Now after the refactoring, the situation is already much better.